### PR TITLE
Drive Base subsystem code implementation and beginning of joystick dr…

### PR DESCRIPTION
**Pull request by SFang-cmd**
…ive commands

### CODE REVIEW

File: src/main/java/frc/robot/oi/XboxUserControls.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    // @Override
+    // public Trigger getIntakeForwardsButton() {
+    // return new Trigger(() -> );
+    // }
+
+    // @Override
+    // public Trigger getIntakeBackwardsButton() {
+    // return new Trigger();
+    // }
```
Comment (hadley31) [inline src/main/java/frc/robot/oi/XboxUserControls.java:88]:
```suggestion

```

File: src/main/java/frc/robot/subsystems/drivetrain/DriveIO.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    /** Sets the velocity of the drivetrain motors (closed loop) */
+    public default void setVelocity(double leftSpeed,
+            double rightSpeed, double leftFFValue, double rightFFValue) {
+    }
+
+    /** Change the mode of the motors */
+    public default void setBrakeMode(boolean enabled) {
+    }
```
Comment (hadley31) [inline src/main/java/frc/robot/subsystems/drivetrain/DriveIO.java:16]:
```suggestion
    public default void setBrakeMode(boolean enabled) {}
```

**Review by hadley31 — CHANGES_REQUESTED** [review]

**Review by hadley31 — DISMISSED** [review]
lgtm

**Review by WAaron-500 — DISMISSED** [review]

**Review by WAaron-500 — DISMISSED** [review]

File: src/main/java/frc/robot/subsystems/flywheel/flyIONonVariable.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import com.ctre.phoenix.motorcontrol.StatorCurrentLimitConfiguration;
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
+// External Imports
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+
+import edu.wpi.first.math.util.Units;
+// Internal Imports
+import frc.robot.Constants;
+
+public class flyIONonVariable implements flyIO {
```
Comment (hadley31) [inline src/main/java/frc/robot/subsystems/flywheel/flyIONonVariable.java]:
```suggestion
public class FlywheelIONonVariable implements FlywheelIO {
```

Comment (hadley31) [reply src/main/java/frc/robot/subsystems/flywheel/flyIONonVariable.java]: Classes and interfaces should have uppercase names (pascal case)

Comment (hadley31) [reply src/main/java/frc/robot/subsystems/flywheel/flyIONonVariable.java]: What is meant by NonVariable?

**Review by hadley31 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/flywheel/flyIONonVariable.java (hunk 3 above)
Comment (WAaron-500) [reply src/main/java/frc/robot/subsystems/flywheel/flyIONonVariable.java]: ah we were planning on having a variable hood shooter some time before dchamps, just setting up the infastructure so that we can change between variable and non variable hoods

Comment (hadley31) [reply src/main/java/frc/robot/subsystems/flywheel/flyIONonVariable.java]: Ah okay, I see

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -47,9 +48,9 @@ private void configureButtonBindings() {
     // Define commands here
     DriveWithJoystick defaultDriveCommand = new DriveWithJoystick(
         drive,
-        controls::getLeftDriveY,
-        controls::getRightDriveX);
-
+        () -> controls.getLeftDriveY(),
+        () -> controls.getRightDriveX());
+
```
Comment (hadley31) [inline src/main/java/frc/robot/RobotContainer.java]: Why this change? Using the lambda is slightly less efficient as it just wraps the function instead of calling it directly

File: src/main/java/frc/robot/Constants.java [hunk 5]
```diff
@@ -20,18 +20,19 @@ import this class (or one of its inner classes)
  */
 public final class Constants {
 
-    public enum BotType {
-        ROBOT_2022_COMP, ROBOT_2022_PRACTICE
-    }
```
Comment (hadley31) [inline src/main/java/frc/robot/Constants.java]: Enum definitions should be at the top of the file

File: src/main/java/frc/robot/RobotContainer.java (hunk 4 above)
Comment (FellStinger) [reply src/main/java/frc/robot/RobotContainer.java]: Didn't realize the efficiency aspect of it, I thought that doing this would be clearer to others what this part is doing, but we can change it back

**Review by hadley31 — APPROVED** [review]
